src/Propulsion.DynamoStore/DynamoStoreSource.fs

```diff
@@ -115,14 +115,27 @@ module private Impl =
 
 [<NoComparison; NoEquality>]
 type LoadMode =
+    /// Skip loading of Datta for events; this is the most efficient mode as it means the Source only needs to read from the index
+    | Minimal of categories : string array
```

Comment:
Naming wise. I'm not sure "Minimal" is descriptive enough for a public facing API. `IndexOnly` comes to mind, but that's too private sounding unless users are expected to know that there's an index and a table (I guess they sort of are expected to understand the underlying storage model)

maybe: `IndexOnly of categories | IndexOnlyFilter of filter | Hydrated of categories | HydratedFilter of filter`

Reply:
Alternative: `IndexOnly | Hydrated` as a simple DU with no params. And `DynamoStoreSource` accepts either `categories` or `categoryFilter` in the constructor arguments

```
type DynamoStoreSource(loadMode, ?categories, ?categoryFilter) =
  let categoryFilter = 
    categories
      |> Option.map (fun xs x -> Array.contains x xs)
      |> Option.orElse categoryFilter
      |> Option.defaultWith (fun _ -> true)
```

Reply:
Great thinking, thanks!

Reply:
Applied; see my other musing...